nerfstudio/data/dataparsers/minimal_dataparser.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 The Nerfstudio Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Data parser for pre-prepared datasets for all cameras, with no additional processing needed"""
+
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Type
+
+import numpy as np
+import torch
+
+from nerfstudio.cameras.cameras import Cameras
+from nerfstudio.data.dataparsers.base_dataparser import (
+    DataParser,
+    DataParserConfig,
+    DataparserOutputs,
+    Semantics,
+)
+from nerfstudio.data.scene_box import SceneBox
+
+
+@dataclass
+class MinimalDataParserConfig(DataParserConfig):
+    """Nerfstudio dataset config"""
+
+    _target: Type = field(default_factory=lambda: MinimalDataParser)
+    """target class to instantiate"""
+    data: Path = Path("/home/nikhil/nerfstudio-main/tests/data/lego_test/minimal_parser")
+
+
+@dataclass
+class MinimalDataParser(DataParser):
+    """Nerfstudio DatasetParser"""
+
+    config: MinimalDataParserConfig
+
+    def _generate_dataparser_outputs(self, split="train"):
+        # pylint: disable=too-many-statements
+        filepath = self.config.data / f"{split}.npz"
+        data = np.load(filepath, allow_pickle=True)
+
+        image_filenames = data["image_filenames"].tolist()
+        mask_filenames = data["mask_filenames"].tolist() if "mask_filenames" in data.keys() else None
+
+        metadata = None
+        if "semantics" in data.keys():
+            semantics = data["semantics"].item()
+            metadata = {
+                "semantics": Semantics(
+                    filenames=semantics["filenames"].tolist(),
+                    classes=semantics["classes"].tolist(),
+                    colors=semantics["colors"].tolist(),
+                    mask_classes=semantics["mask_classes"].tolist(),
+                )
+            }
+
+        scene_box_aabb = torch.from_numpy(data["scene_box"])
+        scene_box = SceneBox(aabb=scene_box_aabb)
+
+        camera_np = data["cameras"].item()
+        cameras = Cameras(
+            fx=torch.from_numpy(camera_np["fx"]),
+            fy=torch.from_numpy(camera_np["fy"]),
+            cx=torch.from_numpy(camera_np["cx"]),
+            cy=torch.from_numpy(camera_np["cy"]),
+            distortion_params=torch.from_numpy(camera_np["distortion_params"])
+            if "distortion_params" in camera_np.keys()
+            else None,
```

Comment:
Make non-inline if, looks a bit odd split between lines.

Reply:
really? This is quite standard for me :P 
Either way this was black - I can try to revert and see if it changes back

Reply:
so black changes it back automatically :P

Reply:
Yea, to clarify, I prefer inline ifs. However, when they go over the line length limit black breaks them up which then defeats the purpose and in my opinion make it more difficult to read.

Reply:
Im still not sure I follow - defeats the purpose of? . This line in particular is over the line length and so gets auto-formatted to the form above

Reply:
This could be a very non-standard preference, but to me I would rate
a)
```
y = a if x else b
```
is better than
b)
```
if x:
   y = a
else
   y = b
```
c)
is better than
```
y = a
   if x
   else b
```
In this case, (a) is not possible since the line is too long causing it to reformat to (c), so I would prefer (b) in this case. Given this is an argument, I would define distortion params earlier. Sorry if this seems very nitpicky

Reply:
ahh I see - sure. done